tests/checks/check-sphinx.fish

```
@@ -0,0 +1,31 @@
+#RUN: fish=%fish %fish %s
+#REQUIRES: sphinx-build --help
```

This test need not depend on fish so it need not be in `tests/checks/` but I guess it's okay.

The advantage of fitting into the `tests/checks/` format is that it's understood by both cmake as well as our standalone test driver, i.e.

```
cargo b && tests/test_driver.py target/debug tests/checks/check-sphinx.fish
```

the alternative would be to make the test a standalone script (and add it to the fish_run_tests target).
I don't think it matters since this is an edge case.

---

I basically just copied the way it is done for checking the po files in `check-translations.fish`. I don't really care about the location, but I think it is good to run the tests locally along with the other tests, so people can spot issues more easily before they hit CI.

I don't like the idea of custom, CMake-dependent handling.

What we could also consider is adding the `-W` (fail on warning) flag in `build.rs`. This would make fish builds fail on sphinx warnings. Without `-E` (fresh env, complete rebuild) the failures would be ignored on subsequent builds, but build times would be longer if we add `-E`.